docs/src/tutorials/performance.jl

```diff
@@ -0,0 +1,158 @@
+# # Performance Tips
+
+# Profile your code! See the [Profiling](../development/profiling.md) page for more details.
```

Comment:
It may be useful to at least summarize generic CUDA optimization guidelines. Maybe something like:

---

Always start by profiling your code (see the [Profiling](../development/profiling.md) page for more details). You first want to analyze your application as a whole, using NSight Systems, in order to:

- get rid of unnecessary memory copies: you don't want to be going back and forth between the CPU and the GPU's memory;
- identifying problematic kernel invocations: you may be launching thousands of kernels which could be fused into a single call;
- finding stalls, where the CPU isn't submitting work fast enough to keep the GPU busy.

If that isn't sufficient, and you identified a kernel that executes slowly, you can try using NSight Compute to analyze that kernel in detail. The tool may prompt you to:

- optimize memory accesses, e.g., avoiding needless global accesses (buffering in shared memory instead), coalescing accesses, etc;
- lowering register pressure so that more threads can be launched on each streaming multiprocessor;
- increase the arithmetic intensity in order for the GPU to be able to hide the latency of memory accesses.

Reply:
I've added some generic tips, hopefully its sufficient if you wouldn't mind taking a look at the changes.

Reply:
Perfect, thanks!